src/main/java/frc/robot/subsystems/ElevatorSubsystem.java

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkFlex;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Configs;
+import frc.robot.Constants.ElevatorConstants;
+
+public class ElevatorSubsystem extends SubsystemBase {
+  public enum ElevatorSetpoint {
+    kElevatorIdle,
+    kElevatorHover,
+    kElevatorL2,
+    kElevatorL3,
+    kElevatorOuttake,
+    kOuttakeElevatorAlgae,
+    kElevatorIntake;
+  }
+
+  // Elevator Motor
+  private SparkFlex elevatorMotor =
+      new SparkFlex(ElevatorConstants.ELEVATOR_MOTOR, MotorType.kBrushless);
+  private RelativeEncoder elevatorEncoder = elevatorMotor.getExternalEncoder();
+
+  private final ProfiledPIDController elevatorPid =
+      new ProfiledPIDController(
+          ElevatorConstants.kP,
+          ElevatorConstants.kI,
+          ElevatorConstants.kD,
+          new TrapezoidProfile.Constraints(
+              ElevatorConstants.ELEVATOR_MAX_VELOCITY,
+              ElevatorConstants.ELEVATOR_MAX_ACCELERATION));
+
+  // Default Current Target
+  private double elevatorSetpoint = ElevatorConstants.ELEVATOR_IDLE_SETPOINT;
+
+  /** Creates a new ElevatorArmSubsystem. */
+  public ElevatorSubsystem() {
+
+    elevatorMotor.configure(
+        Configs.ElevatorConfig.ELEVATOR_CONFIG,
+        ResetMode.kResetSafeParameters,
+        PersistMode.kPersistParameters);
+
+    elevatorEncoder.setPosition(0);
+  }
+
+  public boolean reachedSetpoint() {
+    return Math.abs(elevatorEncoder.getPosition() - elevatorSetpoint)
+        <= ElevatorConstants.ELEVATOR_TOLERANCE;
+  }
+
+  private double getElevatorPosition() {
+    return elevatorEncoder.getPosition();
+  }
+
+  public Command setElevatorResetSpeed() {
```

Comment:
```suggestion
  public Command descendSlowly() {
```
This isn't exactly a setter, use verbs for commands.